### ISSUE TITLE
Problem with commiting to branch with slash in it

### ISSUE THREAD

**Issue by paskal** [github-actions-x/commit#17]
Hello there!

I'm using following code:

```yaml
    - name: commit kernel-specific library
      uses: github-actions-x/commit@v2.6
      with:
        github-token: ${{ secrets.BPFINK_COMMIT_TOKEN }}
        commit-message: 'bump kernel-specific libraries'
        files: pkg/ebpf/*.o
```

Running it on branch paskal/automated_build produces the following [output](https://github.com/bookingcom/bpfink/runs/783950793?check_suite_focus=true#step:10:17):

```
    debug(git(['checkout', '-B', branch]))
  File "/usr/local/lib/python3.7/site-packages/plumbum/commands/base.py", line 96, in __call__
    return self.run(args, **kwargs)[1]
  File "/usr/local/lib/python3.7/site-packages/plumbum/commands/base.py", line 232, in run
    return p.run()
  File "/usr/local/lib/python3.7/site-packages/plumbum/commands/base.py", line 193, in runner
    return run_proc(p, retcode, timeout)
  File "/usr/local/lib/python3.7/site-packages/plumbum/commands/processes.py", line 302, in run_proc
    return _check_process(proc, retcode, timeout, stdout, stderr)
  File "/usr/local/lib/python3.7/site-packages/plumbum/commands/processes.py", line 22, in _check_process
    proc.verify(retcode, timeout, stdout, stderr)
  File "/usr/local/lib/python3.7/site-packages/plumbum/machines/base.py", line 26, in verify
    stderr)
plumbum.commands.processes.ProcessExecutionError: Unexpected exit code: 128
Command line: | /usr/bin/git checkout -B paskal
Stdout:       | M	pkg/ebpf/vfs-3.10.o
Stderr:       | fatal: cannot lock ref 'refs/heads/paskal': 'refs/heads/paskal/automated_build' exists; cannot create 'refs/heads/paskal'
```

It seems like the code of this action relies on not having a slash in the branch name. [Here is the output](https://github.com/bookingcom/bpfink/runs/783983547?check_suite_focus=true#step:10:11) of running the very same code in branch `automated_build`:

```
login paskal
password ***
machine api.github.com
login paskal
password ***

Your branch is up to date with 'origin/automated_build'.
```

**Comment by paskal** [comment]
Pull requests run [suffers from same problem](https://github.com/bookingcom/bpfink/pull/20/checks?check_run_id=784802195#step:10:38):

```
Command line: | /usr/bin/git pull --rebase --autostash origin 20
Stderr:       | fatal: couldn't find remote ref 20
```

**Comment by paskal** [comment]
It's fixed by #18 but new version release is required to make use of it. /cc @rusnasonov.

**Comment by perseus-me** [comment]
done, v2.7 released
